app.ts

```diff
@@ -0,0 +1,171 @@
+//#region AppLogic
+//#region API methods
+function getRecordCount() {
+    return fetch('http://localhost:2050/recordCount')
+        .then((response) => {
+            return response.json();
+        });
+}
+function getColumnNames() {
```

Comment:
This is the more modern approach:
```
async function getUserAsync(name) {
  const response = await fetch(`https://api.github.com/users/${name}`);
  return await response.json();
}
```

Reply:
Agreed, the code feels nice and synchronous if you do this